src/reminder/reminder.service.ts

```diff
@@ -0,0 +1,236 @@
+        const overdue: ListReminderEachDate[] = []
+        for (const date of dateList) {
+            const listReminderEachDate: ModifiedReminder[] = []
+            for (const overdueReminder of overdueReminders) {
+                const overdueDate = new Date(overdueReminder.startingDateTime)
+                overdueDate.setHours(0, 0, 0)
+                if (overdueDate === date) {
+                    listReminderEachDate.push({
+                        rid: overdueReminder.rid,
+                        title: overdueReminder.title,
+                        note: overdueReminder.note,
+                        isRemindCaretaker: overdueReminder.isRemindCaretaker,
+                        importanceLevel: overdueReminder.importanceLevel,
+                        imageid: overdueReminder.imageid,
+                        hour: overdueReminder.startingDateTime.getHours(),
+                        minute: overdueReminder.startingDateTime.getMinutes()
+                    })
+                    overdueReminders.splice(overdueReminders.indexOf(overdueReminder), 1)
+                }
+            }
+            overdue.push({
+                date: date,
+                reminder: listReminderEachDate
+            })
+        }
+
+        // Get Future part
+        const nextWeekDate = new Date(currentDate)
+        nextWeekDate.setDate(nextWeekDate.getDate() + 8)
+        nextWeekDate.setHours(0, 0, 0)
+        currentDate.setMinutes(currentDate.getMinutes() - 1)
+        const futureReminders: Reminder[] = await this.reminderRepository.find({
+            where: {
+                uid: uid,
+                isDone: false,
+                startingDateTime: Between(currentDate, nextWeekDate)
+            },
+            order: {
+                startingDateTime: "ASC"
+            }
+        })
+        currentDate.setMinutes(currentDate.getMinutes() + 1)
+        const tempDate = new Date(currentDate)
+        tempDate.setHours(0, 0, 0)
+        const future: ListReminderEachDate[] = []
+        for (let i = 0; i < 8; i++) {
+            const listReminderEachDate: ModifiedReminder[] = []
+            for (const futureReminder of futureReminders) {
+                if (futureReminder.startingDateTime.getDate() === tempDate.getDate()) {
+                    listReminderEachDate.push({
+                        rid: futureReminder.rid,
+                        title: futureReminder.title,
+                        note: futureReminder.note,
+                        isRemindCaretaker: futureReminder.isRemindCaretaker,
+                        importanceLevel: futureReminder.importanceLevel,
+                        imageid: futureReminder.imageid,
+                        hour: futureReminder.startingDateTime.getHours(),
+                        minute: futureReminder.startingDateTime.getMinutes()
+                    })
+                    futureReminders.splice(futureReminders.indexOf(futureReminder), 1)
+                }
+            }
```

Comment:
Since this block of codes is used multiple time throughout the service, I think it should be separated to a standalone function.

Reply:
U mean splice function?

Reply:
The whole nested for loop